Code/Framework/AzFramework/AzFramework/Quality/QualitySystemComponent.cpp

```diff
@@ -84,6 +91,248 @@ namespace AzFramework
             console->PerformCommand(AZStd::string_view(m_defaultGroupName),
                 { AZ::ConsoleTypeHelpers::ToString(qualityLevel) }, AZ::ConsoleSilentMode::Silent);
         }
+
+        // evaluate device rules on demand so device attributes and rules can be added
+        // at runtime if needed
+        EvaluateDeviceRules();
+    }
+
+    namespace Internal
+    {
+        bool HasMatchingDeviceAttributeRule(DeviceAttributeRegistrarInterface* deviceRegistrar, AZ::SettingsRegistryInterface* registry, AZStd::string_view jsonPath)
+        {
+            bool hasMatchingRule = false;
+
+            auto callback = [&](const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+            {
+                if (visitArgs.m_type != AZ::SettingsRegistryInterface::Type::String)
+                {
+                    // object entries (Lua) are not supported yet
+                    AZ_Warning(
+                        "QualitySystemComponent",
+                        false,
+                        "Skipping device attribute rule object entry '%.*s' that is not a string.",
+                        AZ_STRING_ARG(visitArgs.m_fieldName));
+                    return AZ::SettingsRegistryInterface::VisitResponse::Skip;

```

Comment:
This isn't necessary an object either.
It could be a number or boolean as well.